wgpu-types/src/lib.rs

```diff
@@ -817,11 +850,45 @@ pub struct SamplerDescriptor<'a> {
     pub mipmap_filter: FilterMode,
     pub lod_min_clamp: f32,
     pub lod_max_clamp: f32,
+    #[cfg_attr(feature = "serde", serde(deserialize_with="deserialise_sampler_compare_function::deserialize"))]
     pub compare: Option<&'a CompareFunction>,
```

Comment:
I think this will be unnecessary, given that our upstream header just has an extra Invalid enumeration value - https://github.com/webgpu-native/webgpu-headers/blob/ab1580a154453bab4cc878dd6c05c0ae2d39d9f9/webgpu.h#L156